ui-tests/src/test/java/io/jenkins/plugins/analysis/warnings/WarningsPluginUiTest.java

```diff
@@ -0,0 +1,611 @@
+    }
+
+    private void verifyPmd(final Build build) {
+        build.open();
+        AnalysisSummary pmd = new AnalysisSummary(build, PMD_ID);
+        assertThat(pmd).isDisplayed()
+                .hasTitleText("PMD: 2 warnings")
+                .hasNewSize(0)
+                .hasFixedSize(1)
+                .hasReferenceBuild(1)
+                .hasInfoType(InfoType.ERROR);
+
+        AnalysisResult pmdDetails = pmd.openOverallResult();
+        assertThat(pmdDetails).hasActiveTab(Tab.CATEGORIES)
+                .hasTotal(2)
+                .hasOnlyAvailableTabs(Tab.CATEGORIES, Tab.TYPES, Tab.ISSUES);
+
+        build.open();
+        assertThat(openInfoView(build, PMD_ID))
+                .hasInfoMessages("-> found 1 file",
+                        "-> found 2 issues (skipped 0 duplicates)",
+                        "Issues delta (vs. reference build): outstanding: 2, new: 0, fixed: 1")
+                .hasErrorMessages("Can't create fingerprints for some files:");
+    }
+
+    private void verifyCheckStyle(final Build build) {
+        build.open();
+        AnalysisSummary checkstyle = new AnalysisSummary(build, CHECKSTYLE_ID);
+        assertThat(checkstyle).isDisplayed()
+                .hasTitleText("CheckStyle: 3 warnings")
+                .hasNewSize(3)
+                .hasFixedSize(1)
+                .hasReferenceBuild(1)
+                .hasInfoType(InfoType.ERROR);
+
+        AnalysisResult checkstyleDetails = checkstyle.openOverallResult();
+        assertThat(checkstyleDetails).hasActiveTab(Tab.CATEGORIES)
+                .hasTotal(3)
+                .hasOnlyAvailableTabs(Tab.CATEGORIES, Tab.TYPES, Tab.ISSUES);
+
+        IssuesTable issuesTable = checkstyleDetails.openIssuesTable();
+        assertThat(issuesTable).hasSize(3).hasTotal(3);
+
+        DefaultWarningsTableRow tableRow = issuesTable.getRowAs(0, DefaultWarningsTableRow.class);
+        assertThat(tableRow).hasFileName("RemoteLauncher.java")
+                .hasLineNumber(59)
+                .hasCategory("Checks")
+                .hasType("FinalParametersCheck")
+                .hasSeverity("Error")
+                .hasAge(1);
+
+        build.open();
+        assertThat(openInfoView(build, CHECKSTYLE_ID))
+                .hasInfoMessages("-> found 1 file",
+                        "-> found 3 issues (skipped 0 duplicates)",
+                        "Issues delta (vs. reference build): outstanding: 0, new: 3, fixed: 1")
+                .hasErrorMessages("Can't create fingerprints for some files:");
+    }
+
+    private InfoView openInfoView(final Build build, final String toolId) {
+        return new AnalysisSummary(build, toolId).openInfoView();
+    }
+
+    /**
+     * Tests the build overview page by running two builds that aggregate the three different tools into a single
+     * result. Checks the contents of the result summary.
+     */
+    @Test
+    public void should_aggregate_tools_into_single_result() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The JUnit 4 test method name 'should_aggregate_tools_into_single_result' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=5151082)